docs/ProductionChecklistForScalarDBCluster.md

```diff
@@ -0,0 +1,73 @@
+
+### Backup and restore
+
+You should enable the automatic backup feature and PITR feature in the backend database. For details, see [Set up a database for ScalarDB/ScalarDL deployment](./SetupDatabase.md).
+
+## Production checklist: Client application that uses ScalarDB Cluster
+
+The following is a checklist of recommendations when setting up a client application that is using ScalarDB Cluster in a production environment.
+
+### Transaction manager configuration
+
+You must always run CRUD requests via ScalarDB Cluster. To ensure requests are running properly, check the properties file for your client application and confirm that `scalar.db.transaction_manager=cluster` is configured.
+
+* Recommended for production environments
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with gRPC) --> B(<b>ScalarDB Cluster</b> <br> ScalarDB Library with Consensus Commit) --> C(Underlying storage/database)
+  ```
+
+* Not recommended for production environments (for testing purposes only)
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with Consensus Commit) --> B(Underlying storage/database)
+  ```
+
+### Deployment of the client application when using `direct-kubernetes` client mode
+
+If you use [`direct-kubernetes` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#direct-kubernetes-client-mode), you must deploy your client application on the same Kubernetes cluster as the ScalarDB Cluster deployment.
+
+Also, when using `direct-kubernetes` client mode, you must deploy additional Kubernetes resources to make your client application work properly.  For details, see [Deploy your client application on Kubernetes with `direct-kubernetes` mode](https://github.com/scalar-labs/helm-charts/blob/main/docs/how-to-deploy-scalardb-cluster.md#deploy-your-client-application-on-kubernetes-with-direct-kubernetes-mode).
+
+### Transaction handling
+
+You must make that sure your application always runs [`commit()`](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransaction.html#commit--) or [rollback()](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransaction.html#rollback--) after you [begin()](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransactionManager.html#begin--) a transaction. If the application does not run `commit()` or `rollback()`, your application might experience unexpected issues or read inconsistent data from the backend database.
+
+### Exception handling
+
+You must make sure that your application handles transaction exceptions. For details, see [Handle exceptions](https://github.com/scalar-labs/scalardb/blob/master/docs/api-guide.md#handle-exceptions).
```

Comment:
We also have a document for exception handling for two-phase commit interface:
https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md#handle-exceptions

Furthermore, we also have the same for SQL interface and the gRPC API. But, I'm not sure we should list all the documents here.

Reply:
Ah, I see. Indeed, listing all documents is not a bit good... It might increase maintenance costs and be a bit confusing for users...

However, I don't come up with a good idea at this time. So, first I will list the all documents. This is because the main purpose of this document is to tell users `You should do proper error handling in your application`.

Reply:
I added other exception handling documents in e1f5e9b4de902a03d70e20bfb8493a454ac2a16c.